pom.xml

```diff
@@ -676,8 +747,118 @@
                             </gpgArguments>
                         </configuration>
                     </plugin>
+                    <plugin>
+                        <groupId>org.apache.maven.plugins</groupId>
+                        <artifactId>maven-enforcer-plugin</artifactId>
+                        <executions>
+                            <execution>
+                                <id>enforce-all</id>
+                                <goals>
+                                    <goal>enforce</goal>
+                                </goals>
+                                <configuration>
+                                    <rules>
+                                        <DependencyConvergence/>
+                                        <requireJavaVersion>
+                                            <version>[1.8,12)</version>
+                                        </requireJavaVersion>
+                                        <requireMavenVersion>
+                                            <version>[3.5.0,)</version>
+                                        </requireMavenVersion>
+                                    </rules>
+                                </configuration>
+                            </execution>
+                        </executions>
+                    </plugin>
+                </plugins>
+            </build>
+        </profile>
+        <profile>
+            <id>assembly-hugegraph</id>
```

Comment:
after we move the profile, what's the proper building command?

Before:
`mvn clean install -Papache-release -DskipTests`  for building `apache-release` source/binary package

Now: ?

Reply:
When a profile is activated through the **implicit profile activation** mechanism, the **explicit profile activation mechanism** within the same pom becomes ineffective.

So it's necessary to explicitly specify the profiles:

```bash
mvn clean install -Papache-release,assembly-hugegraph,tar-package -DskipTests
```

> ref: https://maven.apache.org/guides/introduction/introduction-to-profiles.html#explicit-profile-activation

Reply:
> When a profile is activated through the **implicit profile activation** mechanism, the **explicit profile activation mechanism** within the same pom becomes ineffective.
> 
> So it's necessary to explicitly specify the profiles:
> 
> ```shell
> mvn clean install -Papache-release,assembly-hugegraph,tar-package -DskipTests
> ```
> 
> > ref: [maven.apache.org/guides/introduction/introduction-to-profiles.html#explicit-profile-activation](https://maven.apache.org/guides/introduction/introduction-to-profiles.html#explicit-profile-activation)

The relevant scripts and documentation need to be updated later.